src/TeamCloud.Orchestrator/Orchestrations/Azure/AzureResourceGroupDeleteActivity.cs

```diff
@@ -0,0 +1,86 @@
+﻿/**
+ *  Copyright (c) Microsoft Corporation.
+ *  Licensed under the MIT License.
+ */
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.Management.Fluent;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Rest.Azure;
+using TeamCloud.Azure;
+using TeamCloud.Model.Data;
+
+namespace TeamCloud.Orchestrator.Orchestrations.Azure
+{
+    public class AzureResourceGroupDeleteActivity
+    {
+        private readonly IAzureSessionService azureSessionService;
+
+        public AzureResourceGroupDeleteActivity(IAzureSessionService azure)
+        {
+            this.azureSessionService = azure ?? throw new ArgumentNullException(nameof(azure));
+        }
+
+        [FunctionName(nameof(AzureResourceGroupDeleteActivity))]
+        public async Task RunActivity(
+            [ActivityTrigger] AzureResourceGroup azureResourceGroup)
+        {
+            if (azureResourceGroup == null)
+                throw new ArgumentNullException(nameof(azureResourceGroup));
+
+            try
+            {
+                var azure = azureSessionService.CreateSession(azureResourceGroup.SubscriptionId);
+                await DeleteResourceGroupLocksAsync(azureResourceGroup, azure).ConfigureAwait(false);
+                await azure.ResourceGroups
+                    .DeleteByNameAsync(azureResourceGroup.ResourceGroupName)
+                    .ConfigureAwait(false);
+            }
+            catch (CloudException ex) when (ex.Message.Contains("could not be found", StringComparison.InvariantCultureIgnoreCase)) 
```

Comment:
The [`CloudException`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.rest.azure.cloudexception?view=azure-dotnet)'s [`Body`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.rest.azure.cloudexception.body?view=azure-dotnet#Microsoft_Rest_Azure_CloudException_Body) property is of type [`CloudError`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.rest.azure.clouderror?view=azure-dotnet).  You should be able to check the [`CloudError.Code`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.rest.azure.clouderror.code?view=azure-dotnet#Microsoft_Rest_Azure_CloudError_Code) instead.

Reply:
Fixed in latest push.